public/pages/Rules/components/RuleEditor/DetectionVisualEditor.tsx

```diff
@@ -192,6 +223,13 @@ export class DetectionVisualEditor extends React.Component<
             }-${selectionIdx}-${dataIdx}`,
           });
         });
+      } else if (typeof selectionMapJSON === 'string' || typeof selectionMapJSON === 'number') {
```

Comment:
As I don't have much context into these changes, it's tough for me to wrap my head around these various `is/else` cases. Not blocking, but it would be helpful to have some comments that elaborate on these cases for future reference.

Reply:
Added comments explaining the cases each condition solves for